### PR TITLE
Fix/cm 217

### PR DESCRIPTION
**Pull request by mgcouchbase**
#### Fixes #.

- [x] Ran `flake8`
- [x] Ran `run_repo_tests.sh`

#### Changes proposed in this pull request:
regular SWG Sanity 
-http://uberjenkins.sc.couchbase.com:8080/view/04%20sync-gateway/job/cen7-sync-gateway-tests-sanity/157/ -- Master
-http://uberjenkins.sc.couchbase.com:8080/job/sync-gateway-tests-ipv6/lastCompletedBuild/testReport/

CBL:
  http://uberjenkins.sc.couchbase.com:8080/view/04%20sanity/job/Net-windows-TestSever-Funtional-tests-sanity2/24/. --Branch 

Topology:   http://uberjenkins.sc.couchbase.com:8080/job/cen7-sync-gateway-functional-tests-topology-cc/64/. -- cm branch
http://uberjenkins.sc.couchbase.com:8080/job/cen7-sync-gateway-functional-tests-topology-cc/63/. --  cm branch

Upgrade Job :
 http://uberjenkins.sc.couchbase.com:8080/job/sync-gateway-upgrade-centos-testserver/483/testReport/.  ---Master
http://uberjenkins.sc.couchbase.com:8080/job/sync-gateway-upgrade-centos-testserver/486/. -- Branch

### CODE REVIEW

File: resources/x509_cert_gen/gen_keystore.sh [hunk 1]
```diff
@@ -59,27 +59,38 @@ arr_hosts=( $hosts )
...
+
+  if  [[ ${host:1:1} == "[" ]]
+  then
+      ip=`echo $host|sed "s/.*\[//;s/\].*//;"`
+      ip="[${ip}]"
+      new_ip=`echo $ip|sed "s/.*\[//;s/\].*//;"`
+      ${SSH} root@${new_ip} "mkdir ${INBOX}" 2>/dev/null || true
+      ${SCP} chain.pem root@${ip}:${INBOX}
+      ${SCP} pkey.key root@${ip}:${INBOX}
+      ${SSH} root@${new_ip} "chmod 777 ${INBOX}${CHAIN}"
```
Comment (sridevi-15) [inline resources/x509_cert_gen/gen_keystore.sh:73]: chmod 777 is not recomended. Is it required for ipv6 VMs?

File: resources/x509_cert_gen/gen_keystore.sh [hunk 2]
```diff
@@ -59,27 +59,38 @@ arr_hosts=( $hosts )
...
+      ${SSH} root@${new_ip} "chmod 777 ${INBOX}${NODE}.key"
+      ip="${CLUSTER}"
+  else
+      ip=`echo $host|sed 's/\"\([^:]*\):.*/\1/'`
+      ${SSH} root@${ip} "mkdir ${INBOX}" 2>/dev/null || true
```
Comment (sridevi-15) [inline resources/x509_cert_gen/gen_keystore.sh:78]: line 78 to 82 are same steps for both if and else. Is it possible to have common steps for both and have if additional logic on if?

File: resources/x509_cert_gen/openssl-san.cnf [hunk 3]
```diff
@@ -1,15 +1,13 @@
...
-
-# Extensions to add to a certificate request
-
 basicConstraints = CA:true
 keyUsage = nonRepudiation, digitalSignature, keyEncipherment
 subjectAltName = @alt_names
 
-[alt_names]
+[ alt_names ]
+DNS.1 = s10501-ip6.qe.couchbase.com
```
Comment (sridevi-15) [inline resources/x509_cert_gen/openssl-san.cnf:13]: you are not supposed to hard code the alt_names. Also I see spaces added in line #1  and #5. if spaces are not required, can you avoid checking this file ? If required, just remove the hard coded VM under alt_names and commit

Comment (sridevi-15) [reply resources/x509_cert_gen/openssl-san.cnf:13]: Also, can you run our regular regession jobs against ipv4 and make sure our current regression is broken.
Just run jobs against ipv4 for
SGW. functionaal tests: sanity is fine 
SGW upgrate test - test server 
SGW topology tests
Any one CBL functional tests: sanity is fine

One SGW functional job which you already ran against ipv6

File: resources/x509_cert_gen/gen_keystore.sh (hunk 1 above)
Comment (mgcouchbase) [reply resources/x509_cert_gen/gen_keystore.sh:73]: yes it is required .

File: resources/x509_cert_gen/gen_keystore.sh (hunk 2 above)
Comment (mgcouchbase) [reply resources/x509_cert_gen/gen_keystore.sh:78]: ip format is different

**Pull request merged**